app/serializers/found_user_with_status_serializer.rb

```diff
@@ -6,5 +6,16 @@ class FoundUserWithStatusSerializer < FoundUserSerializer
   def initialize(object, options = {})
     super
     options[:include_status] = true
+    deprecated
+  end
+
+  private
+
+  def deprecated
+    message =
+      "FoundUserWithStatusSerializer is deprecated. " \
+        "Use FoundUserSerializer with the include_status option instead."
+
+    Discourse.deprecate(message, since: "3.2.0", drop_from: "3.3.0.beta1")
```

Comment:
Since this commit is going into `3.3.0.beta1-dev`and 3.2.0 (stable) is already released, we should likely change this to: 

```suggestion
    Discourse.deprecate(message, since: "3.3.0.beta1", drop_from: "3.3.0.beta2")
```

We can optionally push the `drop_from` version further, especially if you know of existing usage of the deprecated method out in the wild.

Reply:
Oh, that's right, thank you for catching this!

> We can optionally push the `drop_from` version further, especially if you know of existing usage of the deprecated method out in the wild.

3.3.0.beta2 should be fine, there are only two usages, and I'm migrating them in these PRs:

- https://github.com/discourse/discourse-assign/pull/546
- https://github.com/discourse/discourse-circles/pull/63

We mostly need to prevent the new code from starting using this serializer.